sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -1,15 +1,169 @@
 #include "xmp_file_format.h"
 
+#include "xmp_stream.h"
+#include <XMP.incl_cpp>
+#include <XMP.hpp>
+#include <time.h>
+#include <string_utils.h>
+#include <mutex>
+
+using std::string;
+using std::vector;
+
 namespace mip {
 namespace file {
 
+const XMP_StringPtr kMsipNamespace = "http://www.microsoft.com/msip/1.0/";
```

Comment:
I think you need to move all this xmp code to a seperate class.
This class is going to be gigantic with all the read/write and protection code